src/app/sport/[sport]/[competition]/page.tsx

```diff
@@ -0,0 +1,68 @@
+import prisma from '@db/prismaSingleton'
+
+import SelectGrid from '@components/SelectGrid'
+import TopBar from '@components/TopBar'
+
+type PageProps = {
+  params: {
+    competition: string,
+    sport: string,
+  },
+}
+
+const Page = async (props: PageProps) => {
+  const {
+    params: { 
+      competition: competitionKey,
+      sport: sportKey,
+    },
+  } = props
+
+  const sport = await prisma.sport.findUnique({
+    where: {
+      key: sportKey,
+    },
+  })
+  const { id: sportId } = sport || {}
+
+  const competition = await prisma.competition.findUnique({
+    where: {
+      key: competitionKey,
+      sportId: sportId,
+    },
+  })
+  const { id: competitionId, title: competitionTitle } = competition || {}
+
+  const competitionTeams = await prisma.competitionTeam.findMany({
+    where: {
+      competitionId: competitionId,
+    },
+    include: {
+      team: true,
+    }
+  })
+
+  const teams = await prisma.team.findMany()
```

Comment:
with this query i think we can simplify this file quite a bit by removing the queries for sport, competition, and competitionTeams
```
const teams = await prisma.team.findMany({
  where: {
    competitionTeams: {
      some: {
        competition: {
          key: competitionKey,
        },
      },
    },
  },
})
```